src/main/scala/mesosphere/mesos/TaskGroupBuilder.scala

```diff
@@ -142,7 +148,11 @@ object TaskGroupBuilder {
 
     val containerInfo = computeContainerInfo(podDefinition.podVolumes, container)
 
-    builder.setContainer(containerInfo)
+    // Workaround a validation problem with Mesos.
+    // Only set 'ContainerInfo' if it is non-empty.
+    if (podDefinition.podVolumes.nonEmpty || container.image.isDefined) {
```

Comment:
I've changed the code a bit, but we need to make sure, that `ContainerInfo` is only set when some of it's fields are set. Validation rules are much stricter for pods.